src/anomalib/metrics/pimo/utils.py

```diff
@@ -0,0 +1,192 @@
+"""Torch-oriented interfaces for `utils.py`."""
+
+# Original Code
+# https://github.com/jpcbertoldo/aupimo
+#
+# Modified
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+
+import matplotlib as mpl
+import torch
+from torch import Tensor
+
+from . import _validate
+from .enums import StatsOutliersPolicy, StatsRepeatedPolicy
+
+logger = logging.getLogger(__name__)
+
+
+def per_image_scores_stats(
+    per_image_scores: Tensor,
+    images_classes: Tensor | None = None,
+    only_class: int | None = None,
+    outliers_policy: str | StatsOutliersPolicy = StatsOutliersPolicy.NONE.value,
+    repeated_policy: str | StatsRepeatedPolicy = StatsRepeatedPolicy.AVOID.value,
```

Comment:
What are the effects of these outlier policies? And how often do we use them ?

Reply:
if the distribution is "well behaved", you may get rid of it for simplicity, but if the queue(s) of the distribution are very large you probably wanna check it to get a good picture

this last case happened very often in the benchmark
![image](https://github.com/user-attachments/assets/274ac152-1810-4736-b535-bca7ab655885)

---

this util function was proposed to be used as means to select images for qualitative analysis in a non-arbitrary way

it could also be just moved to a tutorial where this use case is show cased